Show a warning if a deprecated region is selected for deploy

ajordens opened the pull request.

ajordens commented on app/scripts/modules/amazon/serverGroup/configure/serverGroupConfiguration.service.js:
```diff
@@ -54,6 +54,15 @@ module.exports = angular.module('spinnaker.aws.serverGroup.configure.service', [
       };
 
+      command.regionIsDeprecated = function() {
+        if (command.backingData === undefined) {
```
Probably a better way to do this ...

ajordens commented:
![image](https://cloud.githubusercontent.com/assets/388652/13019949/65bbf1f2-d187-11e5-9752-0a015907b503.png)

Here it is @anotherchrisberry

anotherchrisberry commented on app/scripts/modules/amazon/serverGroup/configure/serverGroupConfiguration.service.js:
```diff
@@ -54,6 +54,15 @@ module.exports = angular.module('spinnaker.aws.serverGroup.configure.service', [
+
+        var matchedRegion = _.find(command.backingData.filtered.regions, {name: command.region});
+        return matchedRegion ? matchedRegion.deprecated === true : false;
+      };
```
Give this a shot, eh?
```javascript
command.regionIsDeprecated = () => command.backingData && 
  command.backingData.filtered.regions.filter((region) => region.name === command.region && 
    region.deprecated).length;
```

ajordens commented:
Updated PR with your suggestion @anotherchrisberry.

Peter also asked us to clarify that it was the region in the account that was deprecated so I've adjusted the warning message text slightly.

anotherchrisberry commented on app/scripts/modules/core/region/regionSelectField.directive.html:
```diff
@@ -7,7 +7,7 @@
             ng-change="onChange()"
             required>
-      <option ng-repeat="region in regions" value="{{region.name}}" ng-selected="component[field] === region.name">{{region.name}}</option>
+      <option ng-repeat="region in regions" value="{{region.name}}" ng-selected="component[field] === region.name">{{region.name}} {{region.deprecated ? '(deprecated)' : ''}}</option>
```
Do we have the account name available here? Might be nice to include it, i.e. `(deprecated in test)`

No big deal if not.
